tests/test_crud_data.py

```diff
@@ -0,0 +1,373 @@
+class CRUDSoftDeleteInit(CRUDSoftDeleteDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, DataSoftDelete, connect_to_mem_db(), CRUDSoftDeleteInit.rules_validator)
+
+
+class DataSoftDeleteToken(Base, SoftDeleteMixin, SoftDeleteTokenMixin):
+    __tablename__ = 'crud_softdel_token_test'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False, default="")
+
+
+class CRUDSoftDeleteTokenInit(CRUDSoftDeleteWithTokenDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, DataSoftDeleteToken, connect_to_mem_db(), CRUDSoftDeleteTokenInit.rules_validator)
+
+
+class TestCrud(unittest.TestCase):
+
+    def test_crud(self):
+        crud = CRUDInit()
+
+        crud.create({'name': 'test_name', 'email': 'abc@def.com', 'active': True})
+
+        data = result_to_dict(crud.get(1))
+        self.assertNotEqual(data, None)
+        self.assertDictEqual(data, {'id': 1, 'name': 'test_name', 'email': 'abc@def.com', 'active': True})
+        self.assertEqual(data['id'], 1)
+        self.assertEqual(data['name'], 'test_name')
+        self.assertEqual(data['email'], 'abc@def.com')
+        self.assertEqual(data['active'], True)
+
+        crud.update(1, {'active': False})
+
+        data = result_to_dict(crud.get(1))
+        self.assertNotEqual(data, None)
+        self.assertDictEqual(data, {'id': 1, 'name': 'test_name', 'email': 'abc@def.com', 'active': False})
+        self.assertEqual(data['id'], 1)
+        self.assertEqual(data['name'], 'test_name')
+        self.assertEqual(data['email'], 'abc@def.com')
+        self.assertEqual(data['active'], False)
+
+        crud.delete(1)
+
+        with self.assertRaises(StatusCodeException):
+            crud.get(1)
+
+    def test_crud_multiple(self):
+        crud = CRUDInit()
+
+        crud.create({'name': 'Jon', 'email': 'jon@def.com', 'active': True})
+        crud.create({'name': 'Ron', 'email': 'ron@def.com', 'active': True})
+        crud.create({'name': 'Sam', 'email': 'sam@def.com', 'active': True})
+        crud.create({'name': 'Jinny', 'email': 'jinny@def.com', 'active': True})
+        crud.create({'name': 'Rosa', 'email': 'rosa@def.com', 'active': True})
+
+        data = result_to_dict(crud.get(1))
+        self.assertNotEqual(data, None)
+        self.assertDictEqual(data, {'id': 1, 'name': 'Jon', 'email': 'jon@def.com', 'active': True})

```

Comment:
it is possible just to pput the created json in a variable. and compate it here?